### PR TITLE
refactor(notebooks): migrate 9 notebooks to new execution API (agent test)

### PR DESCRIPTION
**Pull request by classiqdor**
## Summary
- Agent-driven migration from old `execute()` API to new `sample()` API for 9 notebooks
- Tests automated migration workflow with dedicated agent instructions
- Validates agent judgment calls on edge cases (statevector notebooks, reserved names, VQE patterns)

## Changes by notebook

| Notebook | Key Changes |
|----------|-------------|
| quantum_autoencoder | `num_shots` → `NUM_SHOTS` naming convention |
| qmc_user_defined | `execute()` → `sample()`, `phase` → `phase_` (reserved name workaround) |
| qsvt_fixed_point_amplitude | `execute()` → `sample()`, struct column flattening (`state.a`, etc.) |
| hidden_shift | `execute()` → `sample()`, `counts_of_output()` → DataFrame iteration |
| discrete_poisson_solver | `execute()` → `calculate_state_vector()` (correctly identified statevector need) |
| dqi_max_xorsat | `execute()` → `sample()`, introduced `NUM_SHOTS` constant |
| qfold | `execute()` → `sample()` |
| high_level_modeling_flexible_qpe | `execute()` → `sample()` |
| oracles_workshop | `execute()` → `sample()` (statevector cells correctly left unchanged) |

## Test plan
- [x] Run tests for all 9 modified notebooks
- [x] 7/9 pass, 1 skip (qml_with_classiq_guide uses `minimize()`, correctly skipped by agent)
- [ ] `test_dqi_max_xorsat` fails on circuit depth assertion (unrelated to migration - synthesis variation)
- [x] Verify no remaining `execute(` calls in migrated notebooks

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: algorithms/amplitude_amplification_and_estimation/qsvt_fixed_point_amplitude_amplification/qsvt_fixed_point_amplitude_amplification.ipynb [hunk 1]
```diff
@@ -42,7 +42,7 @@
   },
```
Comment (TomerGoldfriend) [inline algorithms/amplitude_amplification_and_estimation/qsvt_fixed_point_amplitude_amplification/qsvt_fixed_point_amplitude_amplification.ipynb:42]: <p>Can we run the cell and see that the output is the same, more or less?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/1713/discussion/#comment-3870521964'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

File: tutorials/workshops/oracle_workshop/oracles_workshop.ipynb [hunk 2]
```diff
@@ -97,7 +97,7 @@
   },
```
Comment (TomerGoldfriend) [inline tutorials/workshops/oracle_workshop/oracles_workshop.ipynb:97]: <p>same here, I do not see the output</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/1713/discussion/#comment-3870521976'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

**Review by TomerGoldfriend — APPROVED** [review]
